plugin/storage/grpc/proto/storage.proto

```diff
@@ -0,0 +1,178 @@
+syntax = "proto3";
+
+package jaeger.storage.v1;
+
+option go_package = "api_v1";
+
+import "gogoproto/gogo.proto";
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/duration.proto";
+import "google/protobuf/empty.proto";
+import "google/rpc/status.proto";
+
+import "model.proto";
+
+// Enable gogoprotobuf extensions (https://github.com/gogo/protobuf/blob/master/extensions.md).
+// Enable custom Marshal method.
+option (gogoproto.marshaler_all) = true;
+// Enable custom Unmarshal method.
+option (gogoproto.unmarshaler_all) = true;
+// Enable custom Size method (Required by Marshal and Unmarshal).
+option (gogoproto.sizer_all) = true;
+// Enable registration with golang/protobuf for the grpc-gateway.
+option (gogoproto.goproto_registration) = true;
+
+message GetDependenciesRequest {
+    google.protobuf.Timestamp end_timestamp = 1 [
+      (gogoproto.stdtime) = true,
+      (gogoproto.nullable) = false
+    ];
+    google.protobuf.Duration lookback = 2 [
+      (gogoproto.stdduration) = true,
+      (gogoproto.nullable) = false
+    ];
+}
+
+message GetDependenciesSuccess {
+    repeated bytes dependencies = 1 [
+      (gogoproto.nullable) = false,
+      (gogoproto.customtype) = "github.com/jaegertracing/jaeger/model.DependencyLink"
+    ];
+}
+
+message GetDependenciesResponse {
+    oneof response {
+        GetDependenciesSuccess success = 1;
+        google.rpc.Status error = 2;
+    }
+}
+
+message WriteSpanRequest {
+    jaeger.api_v2.Span span = 1;
+}
+
+message WriteSpanResponse {
+    oneof response {
+        google.protobuf.Empty success = 1;
+        google.rpc.Status error = 2;
```

Comment:
Sorry, this is not what I meant. This Status object is transported by gRPC natively, we don't need to use it in our response payloads.

Reply:
Are you saying to basically remove the *Response messages and simply return *Success from the rpc functions? I've just had a play doing this and it makes the jaeger plugin code that talks to the generated proto code much cleaner.